src/DIRAC/WorkloadManagementSystem/DB/JobLoggingDB.py

```diff
@@ -56,32 +55,66 @@ def addLoggingRecord(
 
-        try:
+        def _get_date(date):
+            # We need to specify that timezone is UTC because otherwise timestamp
+            # assumes local time while we mean UTC.
             if not date:
-                # Make the UTC datetime string and float
-                _date = datetime.datetime.utcnow()
+                # Make the UTC datetime
+                return datetime.datetime.utcnow()
             elif isinstance(date, str):
                 # The date is provided as a string in UTC
-                _date = TimeUtilities.fromString(date)
+                return TimeUtilities.fromString(date)
             elif isinstance(date, datetime.datetime):
-                _date = date
+                return date
+            else:
+                raise Exception("Incorrect date for the logging record")
+
+        try:
+            if isinstance(date, list):
+                _date = []
+                for d in date:
+                    try:
+                        _date.append(_get_date(d))
+                    except Exception:
+                        self.log.exception("Exception while date evaluation")
+                        _date.append(datetime.datetime.utcnow())
             else:
-                self.log.error("Incorrect date for the logging record")
-                _date = datetime.datetime.utcnow()
+                _date = _get_date(date)
         except Exception:
             self.log.exception("Exception while date evaluation")
-            _date = datetime.datetime.utcnow()
-
-        # We need to specify that timezone is UTC because otherwise timestamp
-        # assumes local time while we mean UTC.
-        epoc = _date.replace(tzinfo=datetime.timezone.utc).timestamp() - MAGIC_EPOC_NUMBER
+            _date = [datetime.datetime.utcnow()]
 
         cmd = (
             "INSERT INTO LoggingInfo (JobId, Status, MinorStatus, ApplicationStatus, "
-            + "StatusTime, StatusTimeOrder, StatusSource) VALUES (%d,'%s','%s','%s','%s',%f,'%s')"
-            % (int(jobID), status, minorStatus, applicationStatus[:255], str(_date), epoc, source[:32])
+            + "StatusTime, StatusTimeOrder, StatusSource) VALUES "
         )
 
+        # if JobID is a list, make a bulk insert
+        if isinstance(jobID, list):
+            for i, _ in enumerate(jobID):
+                epoc = _date[i].replace(tzinfo=datetime.timezone.utc).timestamp() - MAGIC_EPOC_NUMBER
+                cmd = cmd + "(%d,'%s','%s','%s','%s',%f,'%s')," % (
+                    int(jobID[i]),
+                    status[i],
+                    minorStatus[i],
+                    applicationStatus[:255],
+                    str(_date[i]),
+                    epoc,
+                    source[:32],
+                )
+            cmd = cmd[:-1]
+        else:  # else make a single insert
+            epoc = _date.replace(tzinfo=datetime.timezone.utc).timestamp() - MAGIC_EPOC_NUMBER
+            cmd = cmd + "(%d,'%s','%s','%s','%s',%f,'%s')" % (
```

Comment:
Would it not be better to use `executemany`?

Reply:
You mean https://dev.mysql.com/doc/connector-python/en/connector-python-api-mysqlcursor-executemany.html ?

Yes probably it would be cleaner but we never used it in DIRAC and do we really need to start ?

Reply:
Added it/